app/assets/javascripts/components/tabs.js

```diff
@@ -88,12 +88,24 @@
     }
   });
 
-  // Show the first tabPanel
-  $tabs.find('.tabPanel:first').attr('aria-hidden', 'false').show();
-
-  // Set state for the first tabsList li
-  $tabsList.find('li:first').addClass('current').find(' > a').attr({
-    'aria-selected': true,
-    tabindex: 0
-  });
+  var $anchor_current;
+  var stripped_url = document.location.toString().split("#");
+  if (stripped_url.length > 1) {
+    // Does an anchor with the correct id exist with this anchor value?
+    $anchor_current = $('[id="tab-' + stripped_url[1] + '"]:first');
+  }
```

Comment:
Lines 92–96 could be expressed as something like…

```js
if (location.hash.length) {
  $currentAnchor = $('a#tab-' + location.hash.replace('#', ''));
}
```

`location.hash` will return `#foo-bar-biz-baz` or an empty string.